inst/@sym/sym.m

```diff
@@ -953,7 +954,7 @@
 %! syms x
 %! y = 2*x;
 %! a = 42;
-%! myfile = tempname ();
+%! [fd, myfile] = make_temp_file__ (tempdir (), 'octsympy-myfile-');
 %! save (myfile, 'x', 'y', 'a')
 %! clear x y a
 %! load (myfile)
```

Comment:
Agree, should have closed the unused fd here

Reply:
but then this is no better than any other racy way...   i.e., does this offer any benefit over `tempname`?

Reply:
This is better than `tempname` because `tempname` does not choose the filename and create the file in a single step (aka atomic operation). The fd is opened for convenience but it's not essential. The `mktemp` shell command similarly returns a filename instead of a fd, but it's still secure.